apps/aecore/lib/aecore/channel/tx/channel_close_mutal_tx.ex

```diff
@@ -0,0 +1,172 @@
+defmodule Aecore.Channel.Tx.ChannelCloseMutalTx do
+  @moduledoc """
+  Aecore structure of ChannelCloseMutalTx transaction data.
+  """
+
+  @behaviour Aecore.Tx.Transaction
+  alias Aecore.Channel.Tx.ChannelCloseMutalTx
+  alias Aecore.Tx.DataTx
+  alias Aecore.Account.{Account, AccountStateTree}
+  alias Aecore.Chain.Chainstate
+  alias Aecore.Channel.ChannelStateTree
+
+  require Logger
+
+  @typedoc "Expected structure for the ChannelCloseMutal Transaction"
+  @type payload :: %{
+          channel_id: binary(),
+          initiator_amount: non_neg_integer(),
+          responser_amount: non_neg_integer()
+        }
+
+  @typedoc "Reason for the error"
+  @type reason :: String.t()
+
+  @typedoc "Structure that holds specific transaction info in the chainstate."
+  @type tx_type_state() :: ChannelStateTree.t()
+
+  @typedoc "Structure of the ChannelMutalClose Transaction type"
+  @type t :: %ChannelCloseMutalTx{
+          channel_id: binary(),
+          initiator_amount: non_neg_integer(),
+          responder_amount: non_neg_integer()
+        }
+
+  @doc """
+  Definition of Aecore ChannelCloseMutalTx structure
+
+  ## Parameters
+  - channel_id: channel id
+  - initiator_amount: amount that account first on the senders list commits
+  - responser_amount: amount that account second on the senders list commits
+  """
+  defstruct [:channel_id, :initiator_amount, :responder_amount]
+  use ExConstructor
+
+  @spec get_chain_state_name :: :channels
+  def get_chain_state_name, do: :channels
+
+  @spec init(payload()) :: ChannelCloseMutalTx.t()
+  def init(
+        %{
+          channel_id: channel_id,
+          initiator_amount: initiator_amount,
+          responder_amount: responder_amount
+        } = _payload
+      ) do
+    %ChannelCloseMutalTx{
+      channel_id: channel_id,
+      initiator_amount: initiator_amount,
+      responder_amount: responder_amount
+    }
+  end
+
+  @doc """
+  Checks transactions internal contents validity
+  """
+  @spec validate(ChannelCloseMutalTx.t(), DataTx.t()) :: :ok | {:error, String.t()}
+  def validate(%ChannelCloseMutalTx{} = tx, data_tx) do
+    senders = DataTx.senders(data_tx)
+
+    cond do
+      tx.initiator_amount + tx.responder_amount < 0 ->
+        {:error, "#{__MODULE__}: Channel cannot have negative total balance"}
+
+      length(senders) != 2 ->
+        {:error, "#{__MODULE__}: Invalid from_accs size"}
+
+      true ->
+        :ok
+    end
+  end
+
+  @doc """
+  Changes the account state (balance) of both parties and closes channel (drops channel object from chainstate)
+  """
+  @spec process_chainstate(
+          Chainstate.account(),
```

Comment:
There are still several places throughout the codebase, where the `Chainstate.accounts()` is misspelled as `Chainstate.account()`, that could lead to potential mistakes.